src/main/resources/specializations.yml

```diff
@@ -0,0 +1,4 @@
+list:
```

Comment:
1. If you put this configuration to `application.yml` spring will parse it automatically (you can remove yaml reader and jackson yaml dependency and ugly try-catch blocks)
2. When spring parsed configuration, it can inject all values to bean marked as `@ConfigurationProperties` (please see `ClinicInfo` class in my repo). As a result you will get rid of downcasts.